File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        profit = 0
        for i in prices:
            pft = i - buy
            if pft < 0:
                buy = i
            
            profit = max(profit, pft)
        
        return profit

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
 int buy=prices[0];
 int pmax=0;
 for(int i=1;i<pricesSize;i++)
 {
       if(prices[i]<buy)
       buy=prices[i];
       if(pmax<(prices[i]-buy))
       pmax=prices[i]-buy;
 }   return pmax;
}